### PR TITLE
add FIELD JSON

### PR DESCRIPTION
**Pull request by zhenio**
FIELD_TYPE_JSON

### CODE REVIEW

**Comment by adamlofts** [comment]
Thanks for the contribution. Can you add to the tests for insert & query of this column type.

**Comment by zhenio** [comment]
> Thanks for the contribution. Can you add to the tests for insert & query of this column type.

Yes ,I can  add tests column type

File: test/single_test.dart [hunk 1]
```diff
@@ -100,4 +101,18 @@ END
...
+  test('json type test', () async {
+    await conn.query('DROP TABLE IF EXISTS tjson');
+    await conn.query('CREATE TABLE tjson(a int, b json NULL)');
+    MySqlClientError e;
+    try {
+      await conn.query(
+          'INSERT INTO `tjson` (a, b) VALUES (?, ?)', [3, '{"test":"test"}']);
+      var result = await conn.query('SELECT * FROM tjson');
+      print(result.first.fields);
+    } on MySqlClientError catch (e1) {
```
Comment (adamlofts) [inline test/single_test.dart:114]: What is the purpose of the try-catch?

File: test/unit/prepared_statements_test.dart [hunk 2]
```diff
@@ -186,9 +189,17 @@ void main() {
       expect(value, equals(13.93));
     });
 
+    //Test
```
Comment (adamlofts) [inline test/unit/prepared_statements_test.dart:192]: Why is this commented out?

File: test/single_test.dart (hunk 1 above)
Comment (zhenio) [reply test/single_test.dart:114]: Thank your reply，No purpose ， this is my copy code

File: test/unit/prepared_statements_test.dart (hunk 2 above)
Comment (zhenio) [reply test/unit/prepared_statements_test.dart:192]: This may be due to my negligence

File: test/single_test.dart (hunk 1 above)
Comment (zhenio) [reply test/single_test.dart]: What do I need to modify

Comment (zhenio) [reply test/single_test.dart]: What do I need to modify

File: test/unit/prepared_statements_test.dart (hunk 2 above)
Comment (4stern) [reply test/unit/prepared_statements_test.dart:192]: its done

**Comment by 4stern** [comment]
i think, all is done here?! please merge!

**Pull request merged**